Classes/Controller/TableOfContentsController.php

```diff
@@ -156,103 +304,56 @@ protected function resolveMenuEntry($entry)
     {
-        $this->view->assign('toc', $this->makeMenuArray());
-    }
+        $entryArray = [];
 
-    /**
-     * This builds a menu array for HMENU
-     *
-     * @access public
-     * @return array HMENU array
-     */
-    public function makeMenuArray()
-    {
-        // Load current document.
-        $this->loadDocument($this->requestData);
-        if (
-            $this->document === null
-            || $this->document->getDoc() === null
-        ) {
-            // Quit without doing anything if required variables are not set.
-            return [];
-        } else {
-            if (!empty($this->requestData['logicalPage'])) {
-                $this->requestData['page'] = $this->document->getDoc()->getPhysicalPage($this->requestData['logicalPage']);
-                // The logical page parameter should not appear again
-                unset($this->requestData['logicalPage']);
-            }
-            // Set default values for page if not set.
-            // $this->piVars['page'] may be integer or string (physical structure @ID)
-            if (
-                (int) $this->requestData['page'] > 0
-                || empty($this->requestData['page'])
-            ) {
-                $this->requestData['page'] = MathUtility::forceIntegerInRange((int) $this->requestData['page'],
-                    1, $this->document->getDoc()->numPages, 1);
-            } else {
-                $this->requestData['page'] = array_search($this->requestData['page'], $this->document->getDoc()->physicalStructure);
+        // don't filter if the entry type is collection or search params are empty
+        if ($entry['type'] != 'collection' && is_array($this->filterParams) && !empty($this->filterParams)) {
+            // currently only title filtering is defined
+            // TODO: add more logic here after another fields are defined
+            if (!str_contains($entry['label'], $this->filterParams[0])) {
+                return $entryArray;
             }
-            $this->requestData['double'] = MathUtility::forceIntegerInRange($this->requestData['double'],
-                0, 1, 0);
         }
-        $menuArray = [];
-        // Does the document have physical elements or is it an external file?
+
+        // Set "title", "volume", "type" and "pagination" from $entry array.
+        $entryArray['title'] = !empty($entry['label']) ? $entry['label'] : $entry['orderlabel'];
+        $entryArray['orderlabel'] = $entry['orderlabel'];
+        $entryArray['type'] = Helper::translate($entry['type'], 'tx_dlf_structures', $this->settings['storagePid']);
+        $entryArray['pagination'] = htmlspecialchars($entry['pagination']);
+        $entryArray['doNotLinkIt'] = 1;
+        $entryArray['ITEM_STATE'] = 'HEADER';
+
+        if ($entry['children'] == NULL) {
+            $entryArray['description'] = $entry['description'];
+            $id = str_replace("LOG", "PHYS", $entry['id']);
```

Comment:
`$entry['id']` holds a XML ID, right? Then you can't rely on them always being `LOG...` and `PHYS...`. This might be the case for METS files produced by Kitodo.Production, but it's not standardized. Instead of assuming ID values here, we should use the actual values taken from the METS file.

But besides that, why is it even necessary to replace the ID value here? If the references in the METS file are pointing to the wrong target, we should fix the METS file rather then implementing a workaround in the code.

Reply:
But the images are stored inside the physical structure and entry["id"] contains id of logical structure. Because of that, the id number is always the same but prefix needs to be changed.

Reply:
Yes, but nevertheless we can't just rely on the IDs always having the same numbering. You can get the physical structure ID related to a given logical structure ID via the array `$this->document->smLinks` (see in `Classes/Common/MetsDocument.php`.

Reply:
Thank you! That's exactly what I was needing here. Changed to read id from smLinks array.